services/galley/schema/src/V37.hs

```diff
@@ -0,0 +1,19 @@
+module V37 (migration) where
+
+import Imports
+import Cassandra.Schema
+import Text.RawString.QQ
+
+migration :: Migration
+migration = Migration 37 "Create table `custom-instances`" $ do
+    schema' [r|
+        CREATE TABLE custom-instances (
+            domain             text,  -- email domain of the custom backend owner.
+            config_json_url    text,  -- url under which this custom backend's
+                                      -- config.json can be got.
+            webapp_welcome_url text,  -- needed for webapp due to CORS trouble
+            block_cloud_users  bool,  -- iff NULL or False, new cloud users
```

Comment:
Let's make this an integer / custom CQL type / custom Haskell type for futureproofing, and maybe rename to `cloud_users`. `cd services/galley && make db-reset` helps to re-create cassandra schemas locally.

Reply:
Thanks, will do!

(I'm not sure we'll need it in this case, and there is this is also sent as a boolean to the clients, so it'll be hard to change anyway.  But yes, custom booleans are never a bad thing.)

@mheinzel what do you think?  here is some pseudo-code:

```haskell
data CustomInstanceAllowCloudUser = CloudUserAllowed | CloudUserBlocked
  deriving (Eq, Ord, Bounded, Enum, Show, Generic)

instance Cql CustomInstanceAllowCloudUser where
    ctype = Tagged IntColumn

    fromCql (CqlInt i) = case i of
        0 -> return CloudUserAllowed
        1 -> return CloudUserBlocked
        n -> fail $ "unexpected relation: " ++ show n
    fromCql _ = fail "custom instance allow cloud user: int expected"

    toCql CloudUserAllowed = CqlInt 0
    toCql CloudUserBlocked = CqlInt 1
```

...  and then we will certainly need to keep the custom json instances...

Reply:
If we want to do future-proofing this way, it would seem only consequential to me also expose it to the clients that way. If we have a JSON instance along the lines of
```
CloudUserAllowed -> "false"
CloudUserBlocked -> "true"
```
and then need to add `CloudUserAllowedOnTuesdays`, what do we send the clients? Will the possible values become `true, false, "on_tuesdays"`?
I'd rather go with `"allowed", "blocked"`, so it's natural to add `"on_tuesdays"`.

Reply:
So, is it conceivable that there will be a third value in the future (not sure what that would be)? If so, I'd suggest to also reflect that in the JSON schema and let the client teams know about it.

Reply:
@jschaul that makes me think that we should stick with the boolean.  i don't see how the question whether users should be blocked can have any other answer but "yes" or "no", and anything we want to do beyond that would just go to a different attribute in json and a different column in cassandra?

generally i think this trick of always using ints for booleans in case a third value pops up works much better if we don't expose the same structure to clients.